src/shared/module-util.c

```diff
@@ -50,10 +87,25 @@ int module_load_and_warn(struct kmod_ctx *ctx, const char *module, bool verbose)
                                          "Inserted module '%s'", kmod_module_get_name(mod));
                         else if (err == KMOD_PROBE_APPLY_BLACKLIST)
                                 log_full(verbose ? LOG_INFO : LOG_DEBUG,
-                                         "Module '%s' is deny-listed", kmod_module_get_name(mod));
+                                         "Module '%s' is deny-listed (by kmod)", kmod_module_get_name(mod));
                         else {
                                 assert(err < 0);
 
+                                if (err == -EPERM) {
+                                        if (!blacklist_parsed) {
+                                                r = proc_cmdline_parse(parse_proc_cmdline_item, NULL, 0);
```

Comment:
Please do not use global variable for `blacklist`, but please declare it in this function, and pass its pointer to `proc_cmdline_parse()`:
```c
int module_load_and_warn(struct kmod_ctx *ctx, const char *module, bool verbose) {
        _cleanup_strv_free_ char **blacklist = NULL;
        bool blacklist_parsed = false;
        ...

                                if (err == -EPERM) {
                                        if (!blacklist_parsed) {
                                                r = proc_cmdline_parse(parse_proc_cmdline_item, &blacklist, 0);
```